dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -663,9 +664,29 @@ protected void handlePublishUnpublish(String entryPath, String newName, boolean
                     publish(true, entryPath);
                 }
             } else {
-                //for workflows method currently doesn't work with --entryname flag
-                errorMessage("Parameter '--entryname' not valid for workflows. See `workflow publish --help` for more information.",
-                    CLIENT_ERROR);
+                try {
+                    final Workflow workflow = workflowsApi.getWorkflowByPath(entryPath, null, false);
```

Comment:
This method is already invoked on line 648. Can't you just use `existingWorkflow`? Note, this is a network request, a relatively expensive call, so you don't want to duplicate this call if unnecessary.

Compiler may be possibly be giving you a variable undefined error, if so, but you should fix the code.

Reply:
Updated. No longer makes the additional unnecessary API call.